examples/atlas-advanced-cluster/global-cluster/main.tf

```diff
@@ -0,0 +1,98 @@
+provider "mongodbatlas" {
+  public_key  = var.public_key
+  private_key = var.private_key
+}
+
+resource "mongodbatlas_advanced_cluster" "cluster" {
+  project_id     = mongodbatlas_project.project.id
+  name           = var.cluster_name
+  cluster_type   = "GEOSHARDED"
+  backup_enabled = true
+
+  replication_specs { # zone n1
+    zone_name  = "zone n1"
+    num_shards = 3 # 3-shard Multi-Cloud Cluster
+
+    region_configs { # shard n1 
+      electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+      }
+      analytics_specs {
+        instance_size = "M10"
+        node_count    = 1
+      }
+      provider_name = "AWS"
+      priority      = 7
+      region_name   = "US_EAST_1"
+    }
+
+    region_configs { # shard n2
+      electable_specs {
+        instance_size = "M10"
+        node_count    = 2
+      }
+      analytics_specs {
+        instance_size = "M10"
+        node_count    = 1
+      }
+      provider_name = "AZURE"
+      priority      = 6
+      region_name   = "US_EAST_2"
+    }
+
+    region_configs { # shard n3
+      analytics_specs {
+        instance_size = "M10"
+        node_count    = 1
+      }
+      provider_name = "GCP"
+      priority      = 0
+      region_name   = "US_EAST_4"
+    }
+  }
+
+  replication_specs { # zone n2
+    zone_name  = "zone n2"
+    num_shards = 2 # 2-shard Multi-Cloud Cluster
+
+    region_configs { # shard n1 
+      electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+      }
+      analytics_specs {
+        instance_size = "M10"
+        node_count    = 1
+      }
+      provider_name = "AWS"
+      priority      = 7
+      region_name   = "EU_WEST_1"
+    }
+
+    region_configs { # shard n2
+      electable_specs {
+        instance_size = "M10"
+        node_count    = 2
+      }
+      analytics_specs {
+        instance_size = "M10"
+        node_count    = 1
+      }
+      provider_name = "AZURE"
+      priority      = 6
+      region_name   = "EUROPE_NORTH"
+    }
+  }
+
+  advanced_configuration {
+    javascript_enabled                   = true
+    oplog_size_mb                        = 999
```

Comment:
just curious , why 999 as input?

Reply:
there is no reason. I was looking for a value different from the default one. Do you have a specific value in mind?

Reply:
no concerns thx